src/uberjob/progress/_simple_progress_observer.py

```diff
@@ -196,26 +192,24 @@ def _run_update_thread(self):
             if output_value is not None:
                 self._output(output_value)
 
-    @contextmanager
-    def _lock_and_make_stale(self):
```

Comment:
I got rid of the _lock_and_make_stale context manager because it had a small performance impact and removing it doesn't really make the code worse.